R/TaskClassif_melanoma.R

```diff
@@ -0,0 +1,128 @@
+#' task
+NULL
+
+# @param path (`character(1)`)\cr
+#   The cache_dir/datasets/melanoma folder
+constructor_melanoma = function(path) {
+  base_url = "https://huggingface.co/datasets/carsonzhang/ISIC_2020_small/resolve/main/"
+
+  compressed_tarball_file_name = "hf_ISIC_2020_small.tar.gz"
+  compressed_tarball_path = file.path(path, compressed_tarball_file_name)
+  curl::curl_download(paste0(base_url, compressed_tarball_file_name), compressed_tarball_path)
+  utils::untar(compressed_tarball_path, exdir = path)
+  file.remove(compressed_tarball_path)
+
+  training_metadata_file_name = "ISIC_2020_Training_GroundTruth_v2.csv"
+  training_metadata = data.table::fread(here::here(path, training_metadata_file_name))
+
+  test_metadata_file_name = "ISIC_2020_Test_Metadata.csv"
+  test_metadata = data.table::fread(here::here(path, test_metadata_file_name))
+
+  training_metadata = training_metadata[, split := "train"]
+  test_metadata = setnames(test_metadata,
+    old = c("image", "patient", "anatom_site_general"),
+    new = c("image_name", "patient_id", "anatom_site_general_challenge")
+  )[, split := "test"]
+  metadata = rbind(training_metadata, test_metadata, fill = TRUE)
+
+  melanoma_ds_generator = torch::dataset(
+    initialize = function() {
+      self$.metadata = metadata
+      self$.path = path
+    },
+    .getitem = function(idx) {
+      force(idx)
+
+      x = torchvision::base_loader(file.path(self$.path, paste0(self$.metadata[idx, ]$file_name)))
+      x = torchvision::transform_to_tensor(x)
+
+      return(list(x = x))
+    },
+    .length = function() {
+      nrow(self$.metadata)
+    }
+  )
+
+  melanoma_ds = melanoma_ds_generator()
+
+  dd = as_data_descriptor(melanoma_ds, list(x = c(NA, 3, 128, 128)))
+  lt = lazy_tensor(dd)
+
+  return(cbind(metadata, data.table(image = lt)))
+}
+
+load_task_melanoma = function(id = "melanoma") {
+  cached_constructor = function(backend) {
+    data = cached(constructor_melanoma, "datasets", "melanoma")$data
+
+    # remove irrelevant cols: image_name, target
+    print(names(data))
+    # if ("image_name" %in% names(data)) data[, image_name := NULL]
+    data[, image_name := NULL]
+    data[, target := NULL]
+
+    # change the encodings of variables: diagnosis, benign_malignant
+    data[, benign_malignant := factor(benign_malignant, levels = c("benign", "malignant"))]
+
+    char_features = c("sex", "anatom_site_general_challenge")
+    data[, (char_features) := lapply(.SD, factor), .SDcols = char_features]
+
+    dt = cbind(
+      data,
+      data.table(
+        ..row_id = seq_len(nrow(data))
+      )
+    )
+
+    DataBackendDataTable$new(data = dt, primary_key = "..row_id")
+  }
+
+  backend = DataBackendLazy$new(
+    constructor = cached_constructor,
+    rownames = seq_len(32701 + 10982),
+    col_info = load_col_info("melanoma"),
+    primary_key = "..row_id"
+  )
+
+  task = TaskClassif$new(
+    backend = backend,
+    id = "melanoma",
+    target = "benign_malignant",
```

Comment:
I would use a shorter name for the target, something like `"outcome"` or somethjing similar